### PR TITLE
Implement thread functions

### PR DESCRIPTION
**Pull request by kalbasit**
based on #6 and will be rebased off of master when #6 is merged.

### CODE REVIEW

File: messages.go [hunk 1]
```diff
@@ -8,9 +8,56 @@ package notmuch
...
+	C.notmuch_messages_move_to_next(ms.cptr)
+	return true
+}
+
+// Tags return a list of tags from all messages.
+//
+// WARNING: You can no longer iterate over messages after calling this
+// function, because the iterator will point at the end of the list.  We do not
+// have a function to reset the iterator yet and the only way how you can
+// iterate over the list again is to recreate the message list.
```
Comment (zenhack) [inline messages.go:38]: The docs in notmuch.h says this returns NULL on error. We should handle that;
right now we'll end up returning a *Tags with a nil cptr. Unfortunately it
doesn't say what sorts of errors can occur and there's no _st variant. Maybe
it's just OOM? we should pin that down.

File: messages.go [hunk 2]
```diff
@@ -8,9 +8,56 @@ package notmuch
...
+	}
+	runtime.SetFinalizer(ts, func(ts *Tags) {
+		C.notmuch_tags_destroy(ts.cptr)
+	})
+	return ts
+}
+
+// Get fetches the currently selected message.
+func (ms *Messages) get() *Message {
+	cmessage := C.notmuch_messages_get(ms.cptr)
```
Comment (zenhack) [inline messages.go:56]: You need to call checkOOM here.

File: tag.go [hunk 3]
```diff
@@ -0,0 +1,20 @@
+package notmuch
+
+// Copyright © 2015 The go.bindings Authors. Authors can be found in the AUTHORS file.
```
Comment (zenhack) [inline tag.go]: I just pushed a patch to master that changes the copyright holder to go.notmuch
globally; would you rebase and fix it here as well?

File: thread.go [hunk 4]
```diff
@@ -26,3 +31,69 @@ func (t *Thread) GetSubject() string {
 func (t *Thread) GetID() string {
 	return C.GoString(C.notmuch_thread_get_thread_id(t.cptr))
 }
+
+// Count returns the total number of messages in the current thread.
+func (t *Thread) Count() int {
+	return int(C.notmuch_thread_get_total_messages(t.cptr))
+}
+
+// CountMatched returns the total number of messages in the current thread.
```
Comment (zenhack) [inline thread.go]: Not quite true, see notmuch.h. Please correct this.

File: thread.go [hunk 5]
```diff
@@ -26,3 +31,69 @@ func (t *Thread) GetSubject() string {
...
+// oldest-first order.
+func (t *Thread) Messages() *Messages {
+	return &Messages{
+		thread: t,
+		cptr:   C.notmuch_thread_get_messages(t.cptr),
+	}
+}
+
+// Authors returns the list of authors.
+func (t *Thread) Authors() []string {
```
Comment (zenhack) [inline thread.go]: Per the docs in notmuch.h, there's a "|" separating matched vs.
non-matched authors. Wee need to deal with that somehow or we'll get
an author like "Bob|Alice" sometimes.

There's also the qeustion as to whether we want to propagate that information
back to the user. We could return ([]string, []string) maybe?

**Comment by zenhack** [comment]
I made a couple comments, but this otherwise looks good.

File: messages.go (hunk 1 above)
Comment (kalbasit) [inline messages.go:38]: how about we make it OOM for now and we add a TODO to pin it down afterwards?

**Comment by kalbasit** [comment]
@zenhack addressed all of your comments. PTAL.

**Comment by zenhack** [comment]
You're still missing the call to checkOOM in Messages.get. Otherwise I'm happy.

**Comment by kalbasit** [comment]
done

**Comment by zenhack** [comment]
Quoting Wael M. Nasreddine (2015-09-28 17:03:44)

>    done

Great, merged.

**Pull request merged**